code/modules/mob/living/simple_animal/friendly/ratking.dm

```diff
@@ -0,0 +1,295 @@
+#define RAT_MAYOR_LEVEL 1
+#define RAT_BARON_LEVEL 3
+#define RAT_DUKE_LEVEL 5
+#define RAT_KING_LEVEL 10
+#define RAT_EMPEROR_LEVEL 20
+#define RAT_SAVIOR_LEVEL 30
+#define RAT_GOD_LEVEL 50
+
+
+/proc/announceToRodents(var/message)
+	for(var/R in world_mouses)
+		R << message
+
+// Since there's no proc for the current object to check crossed objects
+/mob/living/simple_animal/mouse/Crossed(atom/movable/O)
+	..()
+
+	if( !health )
+		return
+
+	if( istype(O,/mob/living/simple_animal/mouse/king))
+		var/mob/living/simple_animal/mouse/king/K = O
+		if( !K.health )
+			return
+
+		src.visible_message("<span class='warning'>[src] joins the [K.swarm_name] of \the [K]</span>", \
+							"<span class='notice'>We join our brethren in \the [K.swarm_name]. Long live \the [K].<span>")
+		K.absorb(src)
+
+/mob/living/simple_animal/mouse/king

```

Comment:
It doesn't seem that friendly, but I guess it makes sense as a mouse child-type.